webpack/ForemanInventoryUpload/Components/AccountList/__tests__/AccountListActions.test.js

```diff
@@ -24,4 +24,21 @@ const fixtures = {
   'should restartProcess': () => restartProcess(accountID, activeTab),
 };
 
-describe('AccountList actions', () => testActionSnapshotWithFixtures(fixtures));
+describe('AccountList actions', () => {
+  beforeEach(() => {
+    API.post.mockImplementation(async () => {});
+  });
+
+  testActionSnapshotWithFixtures(fixtures);
+
+  it('should invoke toast notification upon failure', () => {
+    const apiError = new Error('test error');
+
+    API.post.mockImplementation(
+      () =>
+        new Promise((resolve, reject) => {
+          reject(apiError);
+        })
+    );
+  });
```

Comment:
Actually I found a good example in tasks to test also the action failure: https://github.com/theforeman/foreman-tasks/blob/master/webpack/ForemanTasks/Components/TasksTable/__tests__/TasksBulkActions.test.js#L25

you can add another fixture that will check for failure, something like:

```js
const fixtures = {
  ...,
  'should invoke toast notification upon failure': () => {
    API.post.mockImplementation(() =>
      Promise.reject(new Error('Test Error'))
    );
    return restartProcess(accountID, activeTab)
  }
}
```

Reply:
There is also [mockfnmockImplementationOnce](https://jestjs.io/docs/en/mock-function-api#mockfnmockimplementationoncefn) which might be better if we also want to test the `success` case